lib/internal/Magento/Framework/Mview/View/Subscription.php

```diff
@@ -117,8 +117,19 @@ public function create()
 
             // Add statements for linked views
             foreach ($this->getLinkedViews() as $view) {
+                // Store current column name for reverting back later.
+                $originalColumnName = $this->getColumnName();
```

Comment:
This solution looks like we are trying to hack the subscription logic by changing object property.

Reply:
@lenaorobei I looked at the code again and your comment and it's a bit confusing
1/ It seems we both agree we need to use the subscription column name
2/ I'm not sure why you mention `not the changelog column name`, we need two column names - changelog column is for the changelog table and that's not going to change (even you mentioned same). What we need is to change how we read the subscription column name

My understanding, Subscription is created once for all the subscribers together and whichever subscription we modify last - columnName is used from that and re-used for all the triggers.

I've tried to simplify the code to make it look not (or less) hacky but to me it seems changes are required in the same direction as I've taken or a major rework would be required in how the subscriptions are created (refactor how the statement is built for linked views)

Reply:
You are right, we need both changelog and subscription column names, but `$this->getColumnName()` is changelog column, not the subscription one. 
I recommended to move the logic of getting its name to the `buildStatement` method in order to cover both cases - create and delete.

Reply:
We don't have any object / data available as of now in buildStatement to get the column name of the linked view for which buildStatement is called.

We can pass full view object or they way I've done to pass the subscription name, please advice

Reply:
I thought we can get `view` from the `changelog`.

Reply:
not at-least as per the interface

https://github.com/magento/magento2/blob/2.4-develop/lib/internal/Magento/Framework/Mview/View/ChangelogInterface.php

I may be wrong, please guide if I'm missing something here

Reply:
Got it.
Let's pass the view object then.

Reply:
Updated